### PR TITLE
NormalizeL2 transformation

### PR DESCRIPTION
**Pull request by mbencer**

### CODE REVIEW

File: inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ngraph::pass::NormalizeL2FusionWithAdd::NormalizeL2FusionWithAdd() {
+    auto input = ngraph::pattern::any_input();
+
+    auto exp = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
+    auto pow = std::make_shared<ngraph::opset4::Power>(input, exp);
+    auto axes = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
+    auto reduce_sum = std::make_shared<ngraph::opset4::ReduceSum>(pow, axes);
+    auto sqrt = std::make_shared<ngraph::opset4::Sqrt>(reduce_sum);
+    auto eps_const = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
+    auto sqrt_add_eps = std::make_shared<ngraph::opset4::Add>(sqrt, eps_const);
```
Comment (ilyachur) [inline inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp:75]: Maybe we can join these two passes if check `sqrt_max_eps` or `sqrt_add_eps` types in the callback.
@GlebKazantaev what do you think about it?

**Comment by mitruska** [comment]
I would like to finish this task, but I have no permission to push into this branch. 
I see the comments about scalar check approach and [open question](https://github.com/openvinotoolkit/openvino/pull/1892#discussion_r475317451) about the fusion passes.
So we should merge or close this PR and create a new one to apply any changes.
 @openvinotoolkit/openvino-ngraph-maintainers

File: inference-engine/src/transformations/include/transformations/utils/utils.hpp [hunk 2]
```diff
@@ -53,6 +54,36 @@ inline std::string create_ie_output_name(const ngraph::Output<ngraph::Node>& out
...
 
+template <typename T>
+bool has_constant_value(const std::shared_ptr<ngraph::opset4::Constant>& constant,
+                          const T value,
+                          T epsilon = std::numeric_limits<T>::epsilon()) {
+    if (!constant) {
+        return false;
+    }
+
+    if (shape_size(constant->get_shape()) > 1) {
```
[the remark this answers is not in the document]
Comment (mitruska) [reply inference-engine/src/transformations/include/transformations/utils/utils.hpp:65]: I think this check is not only for a scalar but constant with any shape like {1, 1, 1} that we can also cast to a single value.
So [`shape_size`](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/ngraph/shape.hpp#L69-L79) returns "Number of elements in spanned by a shape".

File: inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        if (!exp_input || !axes_input || !eps_attr) {
+            return false;
+        }
+
+        const bool is_square_pow = op::util::has_constant_value<float>(exp_input, 2.0f);
+        if (!is_square_pow) {
+            return false;
+        }
+        if (shape_size(eps_attr->get_shape()) > 1) {
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp:43]: @lazarevevgeny Am I right that `eps_attr` must be a scalar?

File: inference-engine/src/transformations/include/transformations/utils/utils.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbencer) [reply inference-engine/src/transformations/include/transformations/utils/utils.hpp]: added self-explanatory variable (`is_scalar_or_single_elem`) + used `is_scalar` helper

File: inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (lazarevevgeny) [reply inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp:43]: Yes, it must be a scalar. But here we perform fusing of the sub-graph and it is really does not important whether this input is scalar or a tensor with shape [1]

File: inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/transformations/src/transformations/normalize_l2_fusion.cpp:75]: @GlebKazantaev ping

**Review by lazarevevgeny — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
@ilyachur , @GlebKazantaev , please, take a look.

**Comment by ilyachur** [comment]
> @ilyachur , @GlebKazantaev , please, take a look.

I asked @GlebKazantaev to look at this PR. I think we can join two passes from this PR.

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
LGTM

**Pull request merged**